libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -625,9 +626,12 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 				*arguments.front()->annotation().type,
 				*function.parameterTypes().front(), true
 			);
-			// gas <- gas * !value
-			m_context << Instruction::SWAP1 << Instruction::DUP2;
-			m_context << Instruction::ISZERO << Instruction::MUL << Instruction::SWAP1;
+			if (function.location() != Location::Transfer)
```

Comment:
The reason: `CALL` will stipend `2300` is the value is non-zero. We counter this and provide the stipend if value is zero.